File: pi-dev-backend/src/main/java/cloud/hustler/pidevbackend/entity/Task.java
package cloud.hustler.pidevbackend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode
public class Task {
    @Id
    @GeneratedValue(strategy= GenerationType.UUID)
    private UUID uuid_task;
    private String title;
    private String description;
    @Enumerated(EnumType.STRING)
    private TypeStatus status;
    private LocalDate startDate;
    private LocalDate endDate;



    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "farm_id")
    private Farm farm;





}

File: pi-dev-frontend/src/app/core/models/famrs/task.ts
export interface Task {
    uuid_task?: string;  // Make uuid_task optional
    title: string;
    description: string;
    status: 'TO_DO' | 'IN_PROGRESS' | 'DONE';
    startDate: string | null;  // Allow null for the start date
    endDate: string | null;    // Allow null for the end date
    farm: any;
  }
  